scripts/training/train_metal.py

```diff
@@ -66,32 +60,61 @@ def main(dataset_size, target, model_name):
                 "target": target,
             }
         )
-        mlflow.log_metrics(metrics)
-
-        # TODO: refactor this to a function nmrcraft/analysis/ or nmrcraft/evaluation
-        fig_path = "scratch/"
-        if not os.path.exists(fig_path):
-            os.makedirs(fig_path)
-        cm_path = os.path.join(fig_path, "cm.png")
-        title = r"Confusion matrix, TODO add LaTeX symbols"
-        plot_confusion_matrix(
-            cm,
-            classes=y_labels,
-            title=title,
-            path=cm_path,
-        )
-        roc_path = os.path.join(fig_path, "roc.png")
-        title = r"ROC curve, TODO add LaTeX symbols"
-        plot_roc_curve(
-            fpr, tpr, metrics["roc_auc"], title=title, path=roc_path
-        )
-
-        mlflow.log_artifact(cm_path)
-        mlflow.log_artifact(roc_path)
 
+        if isinstance(y_test, list):  # if target is 1D
+            metrics, cm, fpr, tpr = model_evaluation(
+                best_model, X_test, y_test, y_labels, data_loader
+            )
+
+            title = r"Confusion matrix, TODO add LaTeX symbols"
+            plot_confusion_matrix(
+                cm,
+                classes=data_loader.confusion_matrix_label_adapter(y_labels),
+                title=title,
+                path=get_cm_path(),
+            )
+            # Plot ROC
+            title = r"ROC curve, TODO add LaTeX symbols"
+            plot_roc_curve(
+                fpr, tpr, metrics["roc_auc"], title=title, path=get_roc_path()
+            )
+            # Logging 1D only data
+            mlflow.log_artifact(get_roc_path())
+
+        elif (
+            data_loader.more_than_one_target()
+        ):  # Multidimensional target Array and Multiple targets
+            metrics, cm = model_evaluation_nD(
+                best_model, X_test, y_test, y_labels, data_loader
+            )
+
+            title = r"Confusion matrix, TODO add LaTeX symbols"
+            plot_confusion_matrix(
+                cm,
+                classes=data_loader.confusion_matrix_label_adapter(y_labels),
+                title=title,
+                path=get_cm_path(),
+                full=False,
+                columns_set=data_loader.get_target_columns_separated(),
+            )
+
+        else:  # Multidimensional target Array and single target
+            metrics, cm = model_evaluation_nD(
+                best_model, X_test, y_test, y_labels, data_loader
+            )
+            title = r"Confusion matrix, TODO add LaTeX symbols"
+            plot_confusion_matrix(
+                cm,
+                classes=data_loader.confusion_matrix_label_adapter(y_labels),
+                title=title,
+                path=get_cm_path(),
+            )
+
+        # Logging common data
+        mlflow.log_metrics(metrics)
         mlflow.sklearn.log_model(best_model, "model")
-
         print(f"Accuracy: {metrics['accuracy']}")
+        mlflow.log_artifact(get_cm_path())
```

Comment:
Thanks for cleaning up! We might need to adapt this to the script @strsamue  is writing.